### ISSUE TITLE
Keyboard navigation state is triggered when user is typing in input files

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#27416]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out if relevant)

### Describe the issue:

When users type input fields, this triggers the keyboard navigation state and will display focus outlines. This is most visible on Menu which sets focus on hover.

#### Actual behavior:

After typing in an input field, the focus outline is visible when hovering menu items

#### Expected behavior:

After typing in an input field, the focus outline is not visible when hovering menu items

### If applicable, please provide a codepen repro:
https://codesandbox.io/s/mutable-https-bcgblb?file=/example.tsx

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: (Blocking, High, Normal, Low)

Products/sites affected: (if applicable)

**Comment by ling1726** [comment]
Fix most likely needs to be on keyborg, which detects the keyboard navigation state
